### PR TITLE
Upgrade go version to 1.14.3 and linter version to 1.26.0

### PR DESCRIPTION
**Pull request by mukundansundar**
# Description

Upgrade go version to 1.14 
Upgrade linter version to 1.26.0

[Linters](https://golangci-lint.run/usage/linters/)

For now removed following linters:
[godot](https://github.com/tetafro/godot) --- 90+ files being touched 
[goerr113](https://github.com/Djarvur/go-err113) --- touches so many files (should probably be done as a separate work)
[testpackage](https://github.com/maratori/testpackage) --- changing package of `_test` files might break code 

Some Changes: 
* fixed int32 parsing for ETag
* fixed implicit reference in ranged for loop to indexed, base references 
* fixed a WriteFile permission in a test file to 600 

## Issue reference
#334 
## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by mukundansundar** [comment]
Due to a bug https://github.com/golangci/golangci-lint/issues/1157 in golangci-lint v1.27.0, only upgrading to version 1.26.0 ...

**Comment by mukundansundar** [comment]
Due to GoLangCi bug `https://github.com/golangci/golangci-lint/issues/1157` using 1.26.0

File: .github/workflows/components-contrib.yml [hunk 1]
```diff
@@ -21,11 +21,11 @@ jobs:
     name: Build ${{ matrix.target_os }}_${{ matrix.target_arch }} binaries
     runs-on: ${{ matrix.os }}
     env:
-      GOVER: 1.13.7
+      GOVER: 1.14
       GOOS: ${{ matrix.target_os }}
       GOARCH: ${{ matrix.target_arch }}
       GOPROXY: https://proxy.golang.org
-      GOLANGCI_LINT_VER: v1.23.2
+      GOLANGCI_LINT_VER: v1.26.0
```
Comment (youngbupark) [inline .github/workflows/components-contrib.yml:28]: Can we upgrade to 1.27.0 ? https://github.com/golangci/golangci-lint/releases/tag/v1.27.0

Comment (youngbupark) [reply .github/workflows/components-contrib.yml:28]: oh. I saw your comment.

**Pull request merged**